oauth2_http/javatests/com/google/auth/oauth2/UserCredentialsTest.java

```diff
@@ -699,6 +705,54 @@ public void onFailure(Throwable exception) {
     assertTrue("Should have run onSuccess() callback", success.get());
   }
 
+  @Test
+  public void IdTokenCredentials_SdkClientId_correct() throws IOException {
```

Comment:
I'm not sure I understand the tests: the ability of getting an id token doesn't depend on the Client ID, depends on the scopes requested.

gcloud always requests the scopes to obtain an id token, but the client id would be different if for example auth has been obtained with gcloud auth application-default login.

The requirement for the gcloud client id is to authenticate testing cloud run services.

Reply:
that makes sense, thanks!